src/CodeGen_LLVM.cpp

```diff
@@ -1439,6 +1444,11 @@ void CodeGen_LLVM::visit(const Cast *op) {
         debug(4) << "Emulating cast from " << src << " to " << dst << "\n";
         if ((src.is_float() && src.bits() < 32) ||
             (dst.is_float() && dst.bits() < 32)) {
+            string warn_msg =
+                "(b)float16 type operation is emulated, which is likely to slow down the performance. "
+                "If your target supports native (b)float16 operations, "
+                "it could be improved by adding Target feature to enable it.\n";
+            onetime_warnings.try_emplace(WarningKind::EmulatedFloat16, warn_msg);
```

Comment:
Is the error message itself not a valid key? I.e. could it be a std::set instead of a std::map?

Reply:
It could. I think the downside is the cost of computing key equivalence proportional to the length of string, while we can remove enum WarningKind from the code instead. Do you have any preference?

Reply:
I think this is fine as-is, let's merge it